server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -324,9 +325,18 @@ class DeltaSharedTableKernel(
     )
 
     if (includeFiles) {
-      if (predicateHints.isEmpty && jsonPredicateHints.isEmpty) {
+      if (predicateHints.isEmpty) {
```

Comment:
nit: what if we do a reverse check, throw the exception is this is not empty. 
Then we can put the rest of the logic outside the if clause.

Reply:
I can actually just delete this if statement entirely since the kernel function doesn't get executed if predicateHints.isEmpty since that is being caught in DeltaSharingService.scala. Should I remove the double check here?

Reply:
Alternatively if we want the double check in case the kernel implementation gets called when it shouldn't and we want to throw an error, I should be consistent and do the reverse check on predicateHints, maxFiles, startingVersion, and pageToken

Reply:
either way works. 
I'd prefer let's remove this check and add a comment saying that predicateHints is not supported.